src/NSwag.CodeGeneration.TypeScript/Templates/Client.ProcessResponse.HandleStatusCode.liquid

```diff
@@ -6,8 +6,12 @@ const contentDisposition = response.headers ? response.headers["content-disposit
 {%     else -%}
 const contentDisposition = response.headers ? response.headers.get("content-disposition") : undefined;
 {%     endif -%}
-const fileNameMatch = contentDisposition ? /filename[^;=\n]*=(?:(\\?['"])(.*?)\1|(?:[^\s]+'.*?')?([^;\n]*))/g.exec(contentDisposition) : undefined;
-const fileName = fileNameMatch && fileNameMatch.length > 1 ? decodeURIComponent(fileNameMatch[fileNameMatch.length - 1]) : undefined;
+let fileNameMatch = contentDisposition ? /filename="?([^"]*?)"?(;|$)/g.exec(contentDisposition) : undefined;
+let fileName = fileNameMatch && fileNameMatch.length > 1 ? fileNameMatch[1] : undefined;
+if(!fileName){
+	fileNameMatch = contentDisposition ? /filename[^;=\n]*=(?:(\\?['"])(.*?)\1|(?:[^\s]+'.*?')?([^;\n]*))/g.exec(contentDisposition) : undefined;
+	fileName = fileNameMatch && fileNameMatch.length > 1 ? decodeURIComponent(fileNameMatch[fileNameMatch.length - 1]) : undefined;
```

Comment:
`decodeURIComponent(fileNameMatch[fileNameMatch.length - 1])` would return string 'undefined' if the group happen to be undefined.

You can make it work with the new regex and remove old code that you brought back with this:

`fileName = (fileNameMatch && fileNameMatch.length >= 4) ? fileNameMatch[3] || fileNameMatch[2]) : undefined;`
`fileName = fileName ? decodeURIComponent(fileName) : undefined;`

However, I suggest looping over the string as I explained here: https://github.com/RicoSuter/NSwag/pull/3816#issuecomment-1031502345

That would make it use UTF8 value in case when both non-UTF8 and UTF8 is provided.
`regex.exec` only gets the first match when run once.

Reply:
If you still want to run old code, move it to the end into `if (!fileName) { ... }`

Reply:
Do we have any progress on this? had to revert to we have to revert to 13.15.5 in prod for this.

Reply:
@zyofeng I've proposed working solutions. Waiting for @knoepdan to check it out.
I will probably roll back to 13.15.5 as well for now and disable msbuild generation (I have minimal API).

Reply:
@RicoSuter could you roll back https://github.com/RicoSuter/NSwag/pull/3816 for now?
There are a few important features in the release that has it (e.g. support for minimal API).

Reply:
Hi 
Yes I realize this PullRequest is a quick fix and should not be the final solution. Actually, I wanted to look at your proposal more closely so we that it could be the final solution. Unfortunatly, i didnt have time so far. (maybe this weekend). 

So i created this PR with the following in mind: 
- will always return filename when previously it returned one (no support for special characters)
- will always return a filename when the flawed solution returned one. (support for special character but might fail at some)
So no functionality will be broken. However, as I said, this should not be the finall solution.

Reply:
@knoepdan I think we should either fix it properly or rollback.

Reply:
Rolled back for now so we can find a proper solution here.